pkg/plugin/pr-sanitizer/event_handler.go

```diff
@@ -0,0 +1,102 @@
+	SuccessMessage = "PR title conforms with semantic commit message style."
+	// SuccessDetailsPageName is a name of a documentation page that contains additional status details for success state
+	SuccessDetailsPageName = "pr-sanitizer-success"
+)
+
+// GitHubLabelsEventsHandler is the event handler for the plugin.
+// Implements server.GitHubEventHandler interface which contains the logic for incoming GitHub events
+type GitHubLabelsEventsHandler struct {
+	Client  ghclient.Client
+	BotName string
+}
+
+var (
+	handledPrActions = []string{"opened", "reopened", "edited", "synchronized"}
+	defaultTypes     = []string{"chore", "docs", "feat", "fix", "refactor", "style", "test"}
+)
+
+// HandleEvent is an entry point for the plugin logic. This method is invoked by the Server when
+// events are dispatched from the /hook service
+func (gh *GitHubLabelsEventsHandler) HandleEvent(log log.Logger, eventType github.EventType, payload []byte) error {
+	switch eventType {
+	case github.PullRequest:
+		var event gogh.PullRequestEvent
+		if err := json.Unmarshal(payload, &event); err != nil {
+			log.Errorf("Failed while parsing '%q' event with payload: %q. Cause: %q", github.PullRequest, event, err)
+			return err
+		}
+
+		if !utils.Contains(handledPrActions, *event.Action) {
+			return nil
+		}
+
+		change := scm.RepositoryChange{
+			Owner:    *event.Repo.Owner.Login,
+			RepoName: *event.Repo.Name,
+			Hash:     *event.PullRequest.Head.SHA,
+		}
+		statusContext := github.StatusContext{BotName: gh.BotName, PluginName: ProwPluginName}
+		statusService := ghservice.NewStatusService(gh.Client, log, change, statusContext)
+
+		config := LoadConfiguration(log, change)
+		if gh.HasTitleWithValidType(config, *event.PullRequest.Title) {
+			return statusService.Success(SuccessMessage, SuccessDetailsPageName)
+		} else if ok, prefix := wip.HasWorkInProgressPrefix(*event.PullRequest.Title, wip.LoadConfiguration(log, change)); ok {
+			trimmedTitle := strings.TrimSpace(strings.TrimPrefix(*event.PullRequest.Title, prefix))
+			if gh.HasTitleWithValidType(config, trimmedTitle) {
+				return statusService.Success(SuccessMessage, SuccessDetailsPageName)
+			}
+		}
+		return statusService.Failure(TitleVerificationFailureMessage, TitleVerificationFailureDetailsPageName)
+
+	default:
+		log.Warnf("received an event of type %q but didn't ask for it", eventType)
+	}
+	return nil
+}
+
+// HasTitleWithValidType checks if title prefix conforms with semantic message style
+func (gh *GitHubLabelsEventsHandler) HasTitleWithValidType(config PluginConfiguration, title string) bool {
+	prefixes := defaultTypes
+	if len(config.TypePrefix) != 0 {
+		if config.Combine {
+			prefixes = append(prefixes, config.TypePrefix...)
+		} else {
+			prefixes = config.TypePrefix
+		}
+	}
+
+	for _, prefix := range prefixes {
+		if strings.HasPrefix(strings.ToLower(title), strings.ToLower(prefix)) {
```

Comment:
Missing TrimSpace for Title which will fail this condition for `title : = " :test: configures plugin"`

so this should be`strings.HasPrefix(strings.ToLower(strings.TrimSpace(title)), strings.ToLower(strings.TrimSpace(prefix)))` I would trim both befre checking `hasPrefix`